firedrake/mesh.py

```diff
@@ -337,8 +338,288 @@ def _from_cell_list(dim, cells, coords, comm):
     return plex
 
 
-class MeshTopology(object):
-    """A representation of mesh topology."""
+class AbstractMeshTopology(object, metaclass=abc.ABCMeta):
```

Comment:
An abstract class should require that concrete implementations satisfy the interface. You can require that the concrete classes satisfy this by saying

```python
class Abstract(metaclass=abc.ABCMeta):

    @abc.abstractmethod
    def implement_me(self, args):
       pass

    @property
    @abc.abstractmethod(self):
    def implement_this_property(self):
       pass
```

Now any concrete class is forced to implement these. It looks like you have tried to do this "by hand" by having `NotImplementedError` in various methods.

Reply:
I see. Changes made.